server/gitrest/src/routes/summaries.ts

```diff
@@ -21,81 +24,89 @@ import {
 } from "@fluidframework/server-services-client";
 import { Router } from "express";
 import { Provider } from "nconf";
-import {
-    getExternalWriterParams,
-    IExternalWriterConfig,
-    IRepositoryManager,
-    IRepositoryManagerFactory,
-} from "../utils";
+import { getExternalWriterParams, IRepositoryManager, IRepositoryManagerFactory } from "../utils";
 import { handleResponse } from "./utils";
 
-async function getSummaryBlob(repoManager: IRepositoryManager, sha: string): Promise<IWholeFlatSummaryBlob> {
-    const blob = await repoManager.getBlob(
-        sha,
-    );
-    return {
-        content: blob.content,
-        encoding: blob.encoding === "base64" ? "base64" : "utf-8",
-        id: blob.sha,
-        size: blob.size,
-    };
+interface ISummaryVersion {
+    id: string;
+    treeId: string;
 }
 
-async function getFlatSummary(repoManager: IRepositoryManager, sha: string): Promise<IWholeFlatSummary> {
-    const rawTree = await repoManager.getTree(sha, true /* recursive */);
-    const wholeFlatSummaryTreeEntries: IWholeFlatSummaryTreeEntry[] = [];
-    const wholeFlatSummaryBlobPs: Promise<IWholeFlatSummaryBlob>[] = [];
-    rawTree.tree.forEach((treeEntry) => {
-        if (treeEntry.type === "blob") {
-            wholeFlatSummaryTreeEntries.push({
-                type: "blob",
-                id: treeEntry.sha,
-                path: treeEntry.path,
-            });
-            wholeFlatSummaryBlobPs.push(
-                getSummaryBlob(
-                    repoManager,
-                    treeEntry.sha,
-                ),
-            );
+export class WholeSummaryReadGitManager {
+    constructor(
+        private readonly documentId: string,
+        private readonly repoManager: IRepositoryManager,
```

Comment:
Does the `WholeSummaryReadGitManager` know the `tenantId` through `repoManager`?

Reply:
Not really, but repoManager knows about "repoOwner" and "repoName". I think "repoOwner" is hardcoded in r11s config, and "repoName" is tenantId in the case of R11s. Why do you ask?

Reply:
Oh I was just curious, and also thinking about logging/telemetry when we work on that soon 😄  But as you mentioned, repoName should have the info we need, likely.